common/workunit/workunit.cpp

```diff
@@ -9492,9 +9492,25 @@ IPropertyTree *CLocalWorkUnit::getUnpackedTree(bool includeProgress) const
     return ret.getClear();
 }
 
+IPropertyTree *CLocalWorkUnit::queryIndirectTree() const
+{
+    if (indirectTree)
```

Comment:
Does this need protecting with a critical block?

Reply:
Calls are protected either by the existing workunit crit or the comment:

    // MORE - queryPTree isn't really safe without holding CLocalWorkUnit::crit - really need to move these functions into CLocalWorkunit